### PR TITLE
Project2

### PR DESCRIPTION
**Pull request by MichielStock**
Branch for the main project of the second day: flatland

### CODE REVIEW

**Comment by MichielStock** [comment]
I finished the big part of the notebook (solutions are in the corresponding example file).
Can someone already take a look? 👀

File: examples/odesolver.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#=
```
Comment (Beramos) [inline examples/odesolver.jl:1]: I don't see in which notebook this is included?

File: examples/odesolver.jl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Created on 20/01/2021 10:58:10
```
Comment (Beramos) [inline examples/odesolver.jl:2]: Also here, where is this included. Or are you going to show this?

File: notebooks/day2/flatland.jl [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+type and make sure all the provided functions `corners`, `area`, `move!`, `rotate!`,... work. Using `PlottingRecipes`, you can easily
+plot all your shapes (provided you implemented all the helper functions).
+
+Implementing such shapes can have various exciting applications, such as making a drawing tool or a ray tracer. Our
+end goal is to implement a simulator of a toy statistical physics system. Here, we simulate a system with inert particles, leading to self-organization.
+Our simple rejection sampling algorithm that we will use is computationally very demanding, an ideal case study for Julia!
+
+## Assignments
+
+- [ ] add the correct *inner* constructor to your type;
```
Comment (Beramos) [inline notebooks/day2/flatland.jl:36]: Should I remove this since you show the constructors for each of these types?

Comment (MichielStock) [reply notebooks/day2/flatland.jl]: Removed some stuff, so this is an exercise

File: examples/odesolver.jl (hunk 1 above)
Comment (MichielStock) [reply examples/odesolver.jl:1]: Forgot to put this somewhere. I can recover this for the slides?

**Comment by Beramos** [comment]
Most-certainly

Op za 30 jan. 2021 15:53 schreef Michiel Stock <notifications@github.com>:

**Comment by MichielStock** [comment]
I added a script in 5f99af0 to generate some images that can be used to explain flatland
boundingboxes

File: notebooks/day2/flatland.jl [hunk 4]
```diff
@@ -0,0 +1,875 @@
...
+    return shape
+end
+	
+end
+
+# ╔═╡ d08ab6d0-6246-11eb-08a8-152f9802cdfc
+md"""
+## Plotting 
+
+OK, let's take a look at our shapes! We use `RecipesBase` to allow plotting.
```
Comment (Beramos) [inline notebooks/day2/flatland.jl:593]: @DaanVanHauwermeiren  @MichielStock `RecipedBase` wordt hiervoor al uitgelegd zeker?

Comment (MichielStock) [reply notebooks/day2/flatland.jl:570]: Indien niet, uitleggen in de slides

**Comment by Beramos** [comment]
@MichielStock 
1. how do you see me using the tests?
2. Can you check the intersection question, I feel it is really messy at the moment but I don't know how to improve it.

If we fix the above I can close this

**Comment by MichielStock** [comment]
1. You should not use the tests, these are to check if the code is correct (you can copy them in your own check system though). I am not sure however if importing also imports the `in` and `intersect` functions, even if they are not exported.
2. yes

**Comment by MichielStock** [comment]
> @MichielStock
> 
> 1. how do you see me using the tests?
> 2. Can you check the intersection question, I feel it is really messy at the moment but I don't know how to improve it.
> 
> If we fix the above I can close this

I think it looks quite fine. Not easy to grasp, but we can explain in person if they have questions.

**Pull request merged**